src/Codec/Xlsx/Parser.hs

```diff
@@ -39,23 +41,22 @@ import           Codec.Xlsx.Types.Internal.CustomProperties  as CustomProperties
 import           Codec.Xlsx.Types.Internal.Relationships     as Relationships
 import           Codec.Xlsx.Types.Internal.SharedStringTable
 
-
 -- | Reads `Xlsx' from raw data (lazy bytestring)
 toXlsx :: L.ByteString -> Xlsx
-toXlsx = either (error . show) id . toXlsxEither
+toXlsx = either (error . show) id . toXlsxOrError
 
 data ParseError = InvalidZipArchive
                 | MissingFile FilePath
                 | InvalidFile FilePath
                 deriving (Show, Eq)
 
 -- | Reads `Xlsx' from raw data (lazy bytestring), failing with Left on parse error
-toXlsxEither :: L.ByteString -> Either ParseError Xlsx
-toXlsxEither bs = do
-  ar <- first (const InvalidZipArchive) $ Zip.toArchiveOrFail bs
+toXlsxOrError :: MonadError ParseError m => L.ByteString -> m Xlsx
```

Comment:
@bradediger but why so polymorphic? My last proposal was to stay with `Either` but add a type synonym and use `MonadError` instance for it. Users could use e.g. `hoistEither` to lift `Either`